compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -1511,17 +1511,30 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 ),
             );
         }
+
+        let (span, sugg, post) = if let SuggestionTarget::SimilarlyNamed = suggestion.target
+            && let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(span)
+            && let Some(span) = suggestion.span
+            && suggestion.candidate.as_str().starts_with("_")
+            && snippet == suggestion.candidate.as_str()[1..]
```

Comment:
can probably write this as

```suggestion
            && let Some(suggestion) = suggestion.candidate.as_str().strip_prefix("_")
            && snippet == suggestion
```